geometry/cartesian_product.hpp

```diff
@@ -21,19 +21,30 @@ template<typename LTuple, typename RTuple>
 constexpr auto operator-(LTuple const& left, RTuple const& right);
 
 template<typename Scalar, typename Tuple,
+         typename = std::enable_if_t<!quantities::is_tuple_v<Scalar>>,
          typename = std::enable_if_t<quantities::is_tuple_v<Tuple>>>
 constexpr auto operator*(Scalar const& left, Tuple const& right);
 
 // The extra typename lifts an ambiguity on the definition.
 template<typename Tuple, typename Scalar,
          typename = std::enable_if_t<quantities::is_tuple_v<Tuple>>,
+         typename = std::enable_if_t<!quantities::is_tuple_v<Scalar>>,
          typename = void>
 constexpr auto operator*(Tuple const& left, Scalar const& right);
 
 template<typename Scalar, typename Tuple>
 constexpr auto operator/(Tuple const& left, Scalar const& right);
 
 }  // namespace cartesian_product
+
+namespace polynomial_ring {
+
+template<typename LTuple, typename RTuple,
+         typename = std::enable_if_t<quantities::is_tuple_v<LTuple>>,
+         typename = std::enable_if_t<quantities::is_tuple_v<RTuple>>>
+constexpr auto operator*(LTuple const& left, RTuple const& right);
```

Comment:
Should there be a comment that the cartesian product is interpreted as the coordinates in the monomial basis (or should it say monomial basis in the namespace name)?

Reply:
Added a comment.